src/client.rs

```diff
@@ -1280,6 +1280,89 @@ impl<T: Read + Write> Session<T> {
         .and_then(|lines| parse_id_seq(&lines, &mut self.unsolicited_responses_tx))
     }
 
+    /// The [`SETACL` command](https://datatracker.ietf.org/doc/html/rfc4314#section-3.1)
+    ///
+    /// Modifies the ACLs on the given mailbox for the specified identifier.
+    /// Return [`Error::No`] if the logged in user does not have `a` rights on the mailbox.
+    pub fn set_acl<S1: AsRef<str>, S2: AsRef<str>>(
+        &mut self,
+        mailbox_name: S1,
+        identifier: S2,
```

Comment:
nit: I'd probably just use `impl AsRef<str>` for each of these (and in the other methods).

Reply:
Now, FYI it this does follow the pattern you use for `fetch`,  `uid_fetch`, and `rename`.  but I can do that for these.

Reply:
Let's actually go ahead and change those too to be `impl AsRef<str>`. We should do so now since that's a breaking change.

Reply:
@jonhoo I added a new commit that changes all methods to use impl.